src/services/PromotionService.ts

```diff
@@ -68,15 +70,19 @@ export async function deletePromotion(id: string): Promise<void> {
  * Creates a new promotion.
  *
  * @param promotionDTO - An object containing the fields of the promotion
+ * @param image - An promotion image to upload
  */
-export async function postPromotion(promotionDTO: PostPromotionDTO): Promise<void> {
+export async function postPromotion(promotionDTO: PostPromotionDTO, image?: Image): Promise<void> {
   const url = Routes.PROMOTIONS.POST;
   return axios
     .post(url, promotionDTO)
-    .then(({ data }: AxiosResponse<PostPromotionsResponse>) => {
+    .then(async ({ data }: AxiosResponse<PostPromotionsResponse>) => {
       if (isError<PostPromotionsResponse>(data)) {
         return Promise.reject(data);
       }
+      if (image) {
+        await AmazonS3Service.uploadImage(image, data.id);
```

Comment:
Maybe a tiny comment so we don't forget about how if image upload fails, we still resolve